manifests/charts/base/files/gen-istio-cluster.yaml

```diff
@@ -4735,6 +4716,51 @@ subjects:
     name: istiod-service-account
     namespace: istio-system
 ---
+# Source: base/templates/clusterrole.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: istiod-istio-system
+  namespace: istio-system
+  labels:
+    app: istiod
+    release: istio
+rules:
+# permissions to verify the webhook is ready and rejecting
+# invalid config. We use --server-dry-run so no config is persisted.
+- apiGroups: ["networking.istio.io"]
+  verbs: ["create"]
+  resources: ["gateways"]
+
+# Needed for multicluster secret reading
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "watch", "list"]
+
+# For storing CA secret
```

Comment:
Either merge this with the one above, or add a resourceName [] with the name we use for our self-signed cert ( so the create/update is restricted), otherwise it doesn't make sense to keep the other one.

I would also add a comment "for ingress secret reading"

Reply:
Ah yeah probably should be merged - I had it split because I did have resourceName, but resourceName doesn't work with create anyways so I reverted it